src/builders.ts

```diff
@@ -90,11 +91,33 @@ export class BomBuilder {
     )
   }
 
-  private fetchNpmLs (projectDir: string, process: NodeJS.Process): any {
+  private getNpmCommand (process: NodeJS.Process): string {
     // `npm_execpath` is used by `npm` internally, and is propagated when kicking `npm run-script`
     /* eslint-disable-next-line @typescript-eslint/prefer-nullish-coalescing, @typescript-eslint/strict-boolean-expressions -- need to handle optional empty-string */
-    const command = process.env.npm_execpath || 'npm'
+    let command = process.env.npm_execpath || 'npm'
+
+    if (/npx-cli.js/gi.test(command)) {
+      // `npm` must be used for executing `ls`.
+      // Typically `npm-cli.js` is alongside `npx-cli.js`, as such we attempt to use this and validate it exists.
+      const npmPath = command.replace('npx-cli.js', 'npm-cli.js')
```

Comment:
:x: replace the trailing "npx-cli.js" via `/npx-cli\.js$/

attention: your initial regex-test used case-insensitivity (`i`) -- it might need to apply here, again.